pkg/storage/azure/azureclient/azureclient.go

```diff
@@ -80,50 +80,65 @@ func New(opts *Options) (*Client, error) {
 	}
 	coreOpts.PerCallPolicies = opts.Policies
 	creds := opts.Creds
-	if creds == nil {
-		var err error
-		if strings.TrimSpace(opts.ClientSecret) == "" {
-			options := azidentity.WorkloadIdentityCredentialOptions{
-				ClientOptions: coreOpts,
-				ClientID:      opts.ClientID,
-				TenantID:      opts.TenantID,
-				TokenFilePath: opts.FederatedTokenFile,
-			}
-			creds, err = azidentity.NewWorkloadIdentityCredential(&options)
-			if err != nil {
-				return nil, err
-			}
-		} else {
-			options := azidentity.ClientSecretCredentialOptions{
-				ClientOptions: coreOpts,
-			}
-			creds, err = azidentity.NewClientSecretCredential(
-				opts.TenantID,
-				opts.ClientID,
-				opts.ClientSecret,
-				&options,
-			)
-			if err != nil {
-				return nil, err
-			}
-		}
-
-	}
-
 	coreOpts.Retry = policy.RetryOptions{
 		MaxRetries: -1, // try once
 	}
 
 	return &Client{
-		creds:          creds,
-		clientOpts:     &coreOpts,
-		tagset:         opts.TagSet,
-		subscriptionID: opts.SubscriptionID,
+		creds:      creds,
+		clientOpts: &coreOpts,
+		opts:       opts,
 	}, nil
 }
 
+func (c *Client) getCreds() (azcore.TokenCredential, error) {
+	if c.creds != nil {
+		return c.creds, nil
+	}
+
+	var (
+		err   error
+		creds azcore.TokenCredential
+	)
+
+	if strings.TrimSpace(c.opts.ClientSecret) == "" {
```

Comment:
Note that this condition differs from the one on 4.17+ because the MSI condition had not been backported this far. This is correct and expected.